### ISSUE TITLE
Bug: (n/no-unsupported-features/node-builtins) 

### ISSUE THREAD

**Issue by JustFly1984** [eslint-community/eslint-plugin-n#372]
### Environment

Node version: 22.11.0
pnpm version: 9.12.3
ESLint version: 8.57.1
eslint-plugin-n version: 17.12.0
Operating System: MacOS 15.1

### What rule do you want to report?

n/no-unsupported-features/node-builtins

### Link to Minimal Reproducible Example

no link

### What did you expect to happen?

No error for node rules for window. It should imply that it is a browser code.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

while running eslint in terminal, with latest version eslint-plugin-n@17.12.0:
error  The 'CustomEvent' is still an experimental feature and is not supported until Node.js 23.0.0. The configured version range is '>=22.11.0'  n/no-unsupported-features/node-builtins

```ts
export function createCustomEvent(
  type: string,
  data: CustomEventInit
): CustomEvent {
  return new window.CustomEvent(type, data)
}
```
It is not failing with eslint-plugin-n@17.11.1

**Comment by scagood** [comment]
:thinking: I am not sure you want to use this rule in a browser context. You'd be better off disabling the rule in your browser files.

If I was to guess, this issue is happening because `global`, `globalThis`, and `window` are all referenced in [ReferenceTracker#iterateGlobalReferences](https://eslint-community.github.io/eslint-utils/api/scope-utils.html#tracker-iterateglobalreferences) in the `eslint-utils`. I would need to dig more to be sure, but I think that a config change is more correct here.

---

I would suggest a config like this:

```js
export default [
  {
    files: [ '**/browser/**', '**/node-files/**' ],
    rules: {
      // Common rules.
    },
  },
  {
    files: [ '**/node-files/**' ],
    plugin: { n }, 
    rules: {
      // Node specific rules.
    },
  },
  {
    files: [ '**/browser/**' ],
    rules: {
      // Browser specific rules.
    },
  },
];
```

**Comment by JustFly1984** [comment]
@scagood 
Sorry, I'm not using flat config yet, I'm stuck with 8.57.1 for a while. 
This is happening in the vite/remix app repository.
